### PR TITLE
SfReview Component

### PR DESCRIPTION
**Pull request by dz3n**
# Related issue
https://github.com/DivanteLtd/storefront-ui/issues/252

# Scope of work

- New component SfReview implementation
- Update SfAccordion stories file with fade effect param

# Screenshots of visual changes
sfReview_screenshot

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/component-rules.html) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

**Comment by dz3n** [comment]
Updated components

- SfRating - Added general solution to fix the problem with the rating with computed properties
- SfReview - Removed some computed properties. Changed font size for author
- SfProductCard - Modified v-if for some div tags

**Comment by przspa** [comment]
@dz3n can you resolve conflicts :)

**Comment by dz3n** [comment]
Yeah, sorry. It's better now :)

File: packages/shared/styles/components/SfReview.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+$review__author-font-size: $font-size-extra-big-desktop !default;
+$review__author-font-family: $body-font-family-primary !default;
+$review__author-color: $c-text-primary !default;
+$review__date-text-transform: capitalize !default;
+$review__date-font-weight: $body-font-weight-primary !default;
+$review__date-font-size: $font-size-small-desktop !default;
+$review__date-font-family: $body-font-family-secondary !default;
+$review__date-color: $c-dark-secondary !default;
+$review__message-font-weight: $body-font-weight-primary !default;
+$review__message-font-size: $font-size-big-desktop !default;
```
[the remark this answers is not in the document]
Comment (przspa) [reply packages/shared/styles/components/SfReview.scss]: @filrak I adjust it after merge. @dz3n I tryed used  `$font-size-regular-` for cntent and `$font-size-big-` for heading, for example SfTabs

File: packages/vue/src/components/molecules/SfReview/SfReview.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<section class="sf-review">
+    <!-- @slot Review author. Slot content will replace default <div> tag (bind 'author' string). -->
+    <slot name="author" v-bind="{author}">
+        <div class="sf-review__author">{{ author }}</div>
+    </slot>
+    <!-- @slot Review information. Slot content will replace default <div> tag and its inner <div> tags (bind 'finalRating' computed property- boolean|number, 'finalMaxRating' computed propery- number, 'date' string). -->
+    <slot name="info" v-bind="{finalRating, finalMaxRating, date}">
+        <div class="sf-review__info">
+            <div v-bind:class="{ 'sf-review__rating': finalRating }">
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfReview/SfReview.html]:
```suggestion
            <div :class="{ 'sf-review__rating': finalRating }">
```
Please use short syntax

File: packages/vue/src/components/molecules/SfReview/SfReview.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+              class="sf-review__date" 
+              v-bind:class="{ 'sf-review__display-inline-block': finalRating, 'sf-review__margin-left': finalRating }"
+            >{{ date }}</div>
+        </div>
+    </slot>
+    <!-- @slot Review message. Slot content will replace default <div> tag and its inner <span> and <a> tags (bind 'finalMessage' computed property- string, 'readMoreText' string, 'hideFullText' string). -->
+    <slot name="message" v-bind="{finalMessage, readMoreText, hideFullText}">
+        <div v-if="message">
+            <span class="sf-review__message">{{ finalMessage }} &nbsp;</span> 
+            <a href="#" v-show="showReadMore" @click="showFullMessage" class="sf-review__message_read_more">{{ readMoreText }}</a>
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfReview/SfReview.html]: If you do that, you add "#" to url, maybe just use the button?

File: packages/vue/src/components/molecules/SfReview/SfReview.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            <a href="#" v-show="showHide" @click="hideFullMessage" class="sf-review__message_hide">{{ hideFullText }}</a>
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfReview/SfReview.html]: @dz3n what did you think about that
``` html
<button @click="toggleMessage">{{someText}}</button>
```
and someText be computed label?

File: packages/vue/src/components/molecules/SfReview/SfReview.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    date: {
+      type: String,
+      required: true
+    },
+    /**
+     * Message from the reviewer
+     */
+    message: {
+      type: String,
+      required: true
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfReview/SfReview.js]:
```suggestion
      default: ""
```
default is better, please use it always 👍

File: packages/vue/src/components/molecules/SfReview/SfReview.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    finalMessage() {
+      return this.message.length > this.charLimit && this.showReadMore
+        ? this.message.slice(0, this.charLimit) + "..."
+        : this.message;
+    },
+    finalMaxRating() {
+      return this.maxRating <= 0 ? 1 : this.maxRating;
+    }
+  },
+  created: function() {
```
Comment (przspa) [inline packages/vue/src/components/molecules/SfReview/SfReview.js]: @dz3n maybe you can do that without created function? Should not please use mounted

File: packages/shared/styles/components/SfReview.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (dz3n) [reply packages/shared/styles/components/SfReview.scss]: @przspa scss is updated

**Comment by dz3n** [comment]
@przspa All changes are done

**Pull request merged**